snapcraft/application.py

```diff
@@ -0,0 +1,250 @@
+        """
+        # Set the logging level to DEBUG for all craft-libraries. This is OK even if
+        # the specific application doesn't use a specific library, the call does not
+        # import the package.
+        util.setup_loggers(*self._cli_loggers)
+
+        craft_cli.emit.init(
+            mode=craft_cli.EmitterMode.BRIEF,
+            appname=self.app.name,
+            greeting=f"Starting {self.app.name}",
+            log_filepath=self.log_path,
+            streaming_brief=True,
+        )
+
+        dispatcher = craft_cli.Dispatcher(
+            self.app.name,
+            self.command_groups,
+            summary=str(self.app.summary),
+            extra_global_args=self._global_arguments,
+            # TODO: craft-application should allow setting the default command without
+            # overriding `_get_dispatcher()`
+            default_command=unimplemented.Pack,
+        )
+
+        try:
+            craft_cli.emit.trace("pre-parsing arguments...")
+            # Workaround for the fact that craft_cli requires a command.
+            # https://github.com/canonical/craft-cli/issues/141
+            if "--version" in sys.argv or "-V" in sys.argv:
+                try:
+                    global_args = dispatcher.pre_parse_args(["pull", *sys.argv[1:]])
+                except craft_cli.ArgumentParsingError:
+                    global_args = dispatcher.pre_parse_args(sys.argv[1:])
+            else:
+                global_args = dispatcher.pre_parse_args(sys.argv[1:])
+
+            if global_args.get("version"):
+                craft_cli.emit.ended_ok()
+                print(f"{self.app.name} {self.app.version}")
+                sys.exit(0)
+        except craft_cli.ProvideHelpException as err:
+            print(err, file=sys.stderr)  # to stderr, as argparse normally does
+            craft_cli.emit.ended_ok()
+            sys.exit(0)
+        except craft_cli.ArgumentParsingError as err:
+            print(err, file=sys.stderr)  # to stderr, as argparse normally does
+            craft_cli.emit.ended_ok()
+            sys.exit(64)  # Command line usage error from sysexits.h
+        except KeyboardInterrupt as err:
+            self._emit_error(craft_cli.CraftError("Interrupted."), cause=err)
+            sys.exit(128 + signal.SIGINT)
+        # pylint: disable-next=broad-exception-caught
+        except Exception as err:  # noqa: BLE001
+            self._emit_error(
+                craft_cli.CraftError(
+                    f"Internal error while loading {self.app.name}: {err!r}"
+                )
+            )
+            if os.getenv("CRAFT_DEBUG") == "1":
+                raise
+            sys.exit(70)  # EX_SOFTWARE from sysexits.h
+
+        craft_cli.emit.trace("Preparing application...")
+        self.configure(global_args)
+
+        return dispatcher
+
+
+def main() -> int:
+    """Run craft-application based snapcraft with classic fallback."""
+    for lib_name in (
```

Comment:
With craft-app 1.1 you can use [this](https://github.com/canonical/craft-application/blob/main/craft_application/util/logging.py#L20) instead

Reply:
Done!